pkg/kv/kvserver/store.go

```diff
@@ -1159,8 +1161,15 @@ func NewStore(
 	}
 	s.replRankings = newReplicaRankings()
 
+	var err error
+	s.cordoned, err = replicasToCordon(ctx, eng)
+	if err != nil {
+		// TODO(josh): Is it okay to fatal here in case the read to storage fails?
+		log.Fatalf(ctx, "couldn't determine which replicas if any to cordon: %v", err)
+	}
+
 	s.draining.Store(false)
-	s.scheduler = newRaftScheduler(cfg.AmbientCtx, s.metrics, s, storeSchedulerConcurrency)
+	s.scheduler = newRaftScheduler(cfg.AmbientCtx, s.metrics, s, storeSchedulerConcurrency, s.cordoned)
```

Comment:
With destroyStatus in place the raft sched should not need to know about a list of cordoned ranges in the end.